### ISSUE TITLE
Binding loop not detected at compile time when Image::source-clip-height is involved

### ISSUE THREAD

**Issue by NigelBreslaw** [slint-ui/slint#10342]
### Bug Description

In the attached project run demo.slint and it will crash in live-preview. If you set a height on the MyPanel it will work. The example has no image file to load, but that seems to have nothing to do with the issue as it will crash regardless. 

[bug.zip](https://github.com/user-attachments/files/24405755/bug.zip)

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.15
- Platform/OS: MacOS
- Programming Language:
- Backend/Renderer: live-preview (Skia)

**Comment by ogoffart** [comment]
Reduced testcases

```slint
export component MyPanel {
    fb := Image {
        width: 100%;
        source-clip-width: root.width / 0.531px;
        source-clip-height: root.height / 0.531px;
    }

}

export component MainWindow inherits Window {
    MyPanel {
        x: 30px;
        y: 30px;
        width: 200px;
        // height: 200px;
    }
}
```

Causes `Recursion detected` panic.
